src/main/java/com/robohorse/robopojogenerator/generator/postprocessors/KotlinDataClassPostProcessor.java

```diff
@@ -0,0 +1,183 @@
+ * Please put some info here.
+ *
+ * @author Wafer Li
+ * @since 16/10/24 00:30
+ */
+public class KotlinDataClassPostProcessor extends AbsPostProcessor {
+
+    @Inject
+    public KotlinDataClassPostProcessor() {
+
+    }
+
+
+    @Override
+    public String proceedClassBody(ClassItem classItem, GenerationModel generationModel) {
+
+        proceedImports(classItem);
+
+        StringBuilder       classBodyBuilder = new StringBuilder();
+        Map<String, String> classFields      = classItem.getClassFields();
+
+        for (String objectName : classFields.keySet()) {
+
+            String type       = proceedType(classFields.get(objectName));
+            String annotation = proceedAnnotation(classItem.getAnnotation());
+            String fieldName  = proceedField(objectName);
+
+            classBodyBuilder.append(
+                    classTemplateProcessor
+                            .createKotlinDataClassField(type, fieldName, objectName, annotation));
+        }
+
+        if (classBodyBuilder.length() == 0) {
+            // Kotlin don't allow empty constructor
+            classBodyBuilder.append(ClassTemplate.FIELD_KOTLIN_DOT_DEFAULT);
+        }
+        else {
+            // Remove the last comma
+            classBodyBuilder.deleteCharAt(classBodyBuilder.lastIndexOf(","));
+        }
+
+        return classBodyBuilder.toString();
+    }
+
+
+    private String proceedField(String objectName) {
+
+        String fieldName = objectName;
+
+        if (fieldName.contains("-")) {
+            // Turn to lower case with underscore if it is hyphen
+            fieldName = fieldName.replaceAll("-+", "_");
+        }
+
+        if (fieldName.contains("_")) {
+            fieldName = fieldName.replaceAll("_{2,}", "_");
+            fieldName = CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, fieldName);
+        }
+
+        char fieldNameFirstChar = fieldName.charAt(0);
+        if (Character.isDigit(fieldNameFirstChar)) {
+            // The first char is number
+            fieldName = "_" + fieldName;
+        }
+
+        return fieldName;
+    }
+
+
+    @Override
+    protected String createClassItemText(String packagePath, String imports, String classTemplate) {
+
+        return classTemplateProcessor.createClassItemWithoutSemicolon(packagePath,
+                                                                      imports,
+                                                                      classTemplate);
+    }
+
+
+    private String proceedAnnotation(String annotation) {
+
+        if (annotation != null) {
+            String regex = "\"@(.+?)\"";
+            annotation = annotation.replaceAll(regex, "\"ç$1\"");
+            annotation = annotation.replaceAll("@", "@field:");
+
+            regex = "\"ç(.+?)\"";
+            annotation = annotation.replaceAll(regex, "\"@$1\"");
+
+            return annotation;
+        }
+        return null;
+    }
+
+
+    private String proceedType(String type) {
+
```

Comment:
😂 So, what should I do with this method?

Keep it? Or just use other implementation?